File: pset6/mario/mario.py
from cs50 import get_int

# Request input from user and check within limits
while True:
    rows = get_int("Please enter number of rows: ")
    if rows >= 0 and rows <= 23:
        break

width = rows * 2 + 2

# Iterate from top down using print and concatination
for row in range(1, rows + 1):
    print(" " * ((width//2) - row - 1) + "#" * row + "  " + "#" * row)

File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    //intitialise height outside of allowed values
    int h = -1;

    // while height outside limits, request height from the user
    while (h < 0 || h > 23)
    {
        h = get_int("Height: ");
        // test input: printf("%i\n", h);
    }
    // loop acrross each row (r) h number of times, starting at the top
    for (int r = h; r > 0; r--)
    {
        // On each position p, row print h - r blank, then r#
        for (int p = 0; p < h + 1; p++)
        {
            if (p < r - 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }

}